geonode/services/migrations/0048_service_harvester.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 3.2.4 on 2021-09-20 13:10
+
+from django.db import migrations, models, transaction
+import django.db.models.deletion
+import logging
+logger = logging.getLogger(__name__)
+
+
+@transaction.atomic
+def set_harvester(apps, _):
+    from geonode.services import enumerations
+    from geonode.harvesting.models import Harvester
+    from geonode.harvesting.tasks import update_harvestable_resources
+
+    try:
+        MyModel = apps.get_model('services', 'Service')
+        _services_with_empty_harvesters = MyModel.objects.filter(
+            type__in=(enumerations.WMS, enumerations.GN_WMS),
+            harvester__isnull=True)
+        for _service in _services_with_empty_harvesters:
+            service_harvester = Harvester.objects.create(
+                name=_service.name,
+                default_owner=_service.owner,
+                remote_url=_service.service_url,
```

Comment:
Trying to access Service.service_url throws an exception. This makes the transaction rollback and Harvesters for existing Remote services are not created.